src/components/utils/skills-quiz.jsx

```diff
@@ -0,0 +1,90 @@
+import { useMemo } from 'react';
+import { camelCaseObject } from '@edx/frontend-platform/utils';
+import { connectStateResults } from 'react-instantsearch-dom';
+import { DROPDOWN_OPTION_IMPROVE_CURRENT_ROLE } from '../skills-quiz/constants';
+
+export function withCamelCasedStateResults(WrappedComponent) {
```

Comment:
[consideration] Would it make sense to move this more globally within the app, IFF its not specific to skills quiz in particular?